plugins/hydra_ray_launcher/tests/test_ray_aws_launcher.py

```diff
@@ -0,0 +1,286 @@
+
+    if sha != ray_requirements_sha:
+        log.warning("Ray setup.py changed, we may need new launcher test AMI!")
+    wheels = []
+    for p in plugins_path:
+        wheel = build_plugin_wheel(p, tmpdir)
+        wheels.append(wheel)
+    return wheels
+
+
+def build_plugin_wheel(plugin: str, tmp_wheel_dir: str) -> str:
+    os.chdir(Path("plugins") / plugin)
+    log.info(f"Build wheel for {plugin}, save wheel to {tmp_wheel_dir}.")
+    subprocess.getoutput(
+        f"python setup.py sdist bdist_wheel && cp dist/*.whl {tmp_wheel_dir}"
+    )
+    wheel = subprocess.getoutput("ls dist/*.whl").split("/")[-1]
+    chdir_hydra_root()
+    return wheel
+
+
+def build_core_wheel(tmp_wheel_dir: str) -> str:
+    chdir_hydra_root()
+    subprocess.getoutput(
+        f"python setup.py sdist bdist_wheel && cp dist/*.whl {tmp_wheel_dir}"
+    )
+    wheel = subprocess.getoutput("ls dist/*.whl").split("/")[-1]
+    return wheel
+
+
+def upload_and_install_wheels(
+    tmp_wheel_dir: str,
+    yaml: str,
+    core_wheel: str,
+    plugin_wheels: List[str],
+) -> None:
+    ray_rsync_up(yaml, tmp_wheel_dir + "/", temp_remote_wheel_dir)
+    log.info(f"Install hydra-core wheel {core_wheel}")
+    _run_command(
+        [
+            "ray",
+            "exec",
+            yaml,
+            f"pip install {temp_remote_wheel_dir}{core_wheel}",
+        ]
+    )
+
+    for p in plugin_wheels:
+        log.info(f"Install plugin wheel {p}")
+        _run_command(
+            [
+                "ray",
+                "exec",
+                yaml,
+                f"pip install {temp_remote_wheel_dir}{p}",
+            ]
+        )
+
+
+def get_requirements_sha(path_to_hydra_root: Optional[str] = None) -> str:
+    chdir_hydra_root()
+    if path_to_hydra_root:
+        requirements_path = Path(path_to_hydra_root)
+    else:
+        requirements_path = Path("requirements/requirements.txt")
+    with open(requirements_path, "rb") as f:
+        data = f.read()
+        return hashlib.md5(data).hexdigest()  # nosec
+
+
+@pytest.mark.skipif(sys.platform.startswith("win"), reason=win_msg)  # type: ignore
+def test_discovery() -> None:
+    # Tests that this plugin can be discovered via the plugins subsystem when looking for Launchers
+    assert RayAWSLauncher.__name__ in [
+        x.__name__ for x in Plugins.instance().discover(Launcher)
+    ]
+
+
+@pytest.fixture(scope="module")  # type: ignore
+def manage_cluster() -> Generator[None, None, None]:
+    # first assert the SHA of requirements hasn't changed
+    # if changed, means we need to update test AMI.
+
+    if get_requirements_sha() != hydra_core_requirements_sha:
+        log.warning(
+            "Hydra core requirements changes. This may mean we need to update RAY test AMI."
+        )
+
+    # build all the wheels
+    tmpdir = tempfile.mkdtemp()
+    plugin_wheels = build_ray_launcher_wheel(tmpdir)
+    core_wheel = build_core_wheel(tmpdir)
+    connect_yaml = f"""
```

Comment:
Why as a string?
This would be much better as an OmegaConf DictConfig object built on top of a dictionary.
You can use OmegaConf.save() to dump it to a yaml file.

Reply:
thx! I didn't think of it. Changed it to a DictConfig, much cleaner now (also it helped reduced a lot of duplicate code.)